### PR TITLE
EES-5531 Rename `AppSettingsOptions` to `AppOptions`

### PR DESCRIPTION
**Pull request by ntsim**
This PR simplifies the naming of `AppSettingsOptions` to `AppOptions`. This is less verbose and confusing as it makes it clearer the are options relating to the 'app' itself.

As part of this, all remaining `AppOptions` classes have also been moved into an `Options` namespace in their respective subprojects to be consistent with the admin.

## Related changes

- Changed `AppOptions.Url` properties to be consistent across the project (instead of variations of this).

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/PublicationSubscriptionFunctions.cs [hunk 1]
```diff
@@ -22,14 +22,14 @@ namespace GovUk.Education.ExploreEducationStatistics.Notifier.Functions;
 
 public class PublicationSubscriptionFunctions(
     ILogger<PublicationSubscriptionFunctions> logger,
-    IOptions<AppSettingsOptions> appSettingsOptions,
+    IOptions<AppOptions> appSettingsOptions,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/PublicationSubscriptionFunctions.cs:25]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/ReleaseNotifier.cs [hunk 2]
```diff
@@ -16,13 +16,13 @@ namespace GovUk.Education.ExploreEducationStatistics.Notifier.Functions;
 
 public class ReleaseNotifier(
     ILogger<ReleaseNotifier> logger,
-    IOptions<AppSettingsOptions> appSettingsOptions,
+    IOptions<AppOptions> appSettingsOptions,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/ReleaseNotifier.cs:19]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Repositories/PublicationSubscriptionRepository.cs [hunk 3]
```diff
@@ -9,9 +9,9 @@
 
 namespace GovUk.Education.ExploreEducationStatistics.Notifier.Repositories;
 
-public class PublicationSubscriptionRepository(IOptions<AppSettingsOptions> appSettingsOptions) : IPublicationSubscriptionRepository
+public class PublicationSubscriptionRepository(IOptions<AppOptions> appSettingsOptions) : IPublicationSubscriptionRepository
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Notifier/Repositories/PublicationSubscriptionRepository.cs:12]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Services/ApiSubscriptionTableStorageService.cs [hunk 4]
```diff
@@ -5,6 +5,6 @@
 
 namespace GovUk.Education.ExploreEducationStatistics.Notifier.Services;
 
-internal class ApiSubscriptionTableStorageService(IOptions<AppSettingsOptions> appSettingsOptions)
+internal class ApiSubscriptionTableStorageService(IOptions<AppOptions> appSettingsOptions)
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Notifier/Services/ApiSubscriptionTableStorageService.cs:8]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Services/TokenService.cs [hunk 5]
```diff
@@ -10,13 +10,13 @@
 
 namespace GovUk.Education.ExploreEducationStatistics.Notifier.Services
 {
-    public class TokenService(IOptions<AppSettingsOptions> appSettingsOptions) : ITokenService
+    public class TokenService(IOptions<AppOptions> appSettingsOptions) : ITokenService
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Notifier/Services/TokenService.cs:13]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Swagger/SwaggerConfig.cs [hunk 6]
```diff
@@ -11,7 +11,7 @@ namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Swagger;
 
 public class SwaggerConfig(
     IApiVersionDescriptionProvider provider,
-    IOptions<AppSettingsOptions> appSettingsOptions)
+    IOptions<AppOptions> appSettingsOptions)
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Swagger/SwaggerConfig.cs:14]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/FilterMetaRepository.cs [hunk 7]
```diff
@@ -17,10 +17,10 @@ namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Repos
 
 public class FilterMetaRepository(
     PublicDataDbContext publicDataDbContext,
-    IOptions<AppSettingsOptions> appSettingsOptions,
+    IOptions<AppOptions> appSettingsOptions,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/FilterMetaRepository.cs:20]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/LocationMetaRepository.cs [hunk 8]
```diff
@@ -22,10 +22,10 @@ namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Repos
 
 public class LocationMetaRepository(
     PublicDataDbContext publicDataDbContext,
-    IOptions<AppSettingsOptions> appSettingsOptions,
+    IOptions<AppOptions> appSettingsOptions,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Repository/LocationMetaRepository.cs:25]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Publisher.Tests/Services/PublishingServiceTests.cs [hunk 9]
```diff
@@ -389,7 +389,7 @@ private static PublishingService BuildPublishingService(
             IPublicBlobStorageService? publicBlobStorageService = null,
             IMethodologyService? methodologyService = null,
             IReleaseService? releaseService = null,
-            IOptions<AppSettingsOptions>? appSettingsOptions = null,
+            IOptions<AppOptions>? appSettingsOptions = null,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Publisher.Tests/Services/PublishingServiceTests.cs:392]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/StageReleaseContentFunction.cs [hunk 10]
```diff
@@ -16,11 +16,11 @@ namespace GovUk.Education.ExploreEducationStatistics.Publisher.Functions
 {
     public class StageReleaseContentFunction(
         ILogger<StageReleaseContentFunction> logger,
-        IOptions<AppSettingsOptions> appSettingsOptions,
+        IOptions<AppOptions> appSettingsOptions,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/StageReleaseContentFunction.cs:19]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/PublishingService.cs [hunk 11]
```diff
@@ -20,11 +20,11 @@ public class PublishingService(
         IPublicBlobStorageService publicBlobStorageService,
         IMethodologyService methodologyService,
         IReleaseService releaseService,
-        IOptions<AppSettingsOptions> appSettingsOptions,
+        IOptions<AppOptions> appSettingsOptions,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/PublishingService.cs:23]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs [hunk 12]
```diff
@@ -36,7 +36,7 @@ public class ImporterService : IImporterService
         private const int Stage2RowCheck = 1000;
 
         public ImporterService(
-            IOptions<AppSettingsOptions> appSettingsOptions,
+            IOptions<AppOptions> appSettingsOptions,
```
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs:39]: Might wanna change the parameter name to `appOptions` also here

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/PublicationSubscriptionFunctions.cs (hunk 1 above)
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/PublicationSubscriptionFunctions.cs]: Yep fair point! Updated and for all the other cases as well

**Review by duncan-at-hiveit — APPROVED** [review]

**Pull request merged**